geth/txqueue/txqueue.go

```diff
@@ -183,66 +177,53 @@ func (q *TxQueue) Enqueue(tx *common.QueuedTx) error {
 
 // Get returns transaction by transaction identifier
 func (q *TxQueue) Get(id common.QueuedTxID) (*common.QueuedTx, error) {
-	q.mu.RLock()
-	defer q.mu.RUnlock()
-
-	if tx, ok := q.transactions[id]; ok {
-		return tx, nil
+	tx, ok := q.transactions.get(id)
+	if !ok {
+		return nil, ErrQueuedTxIDNotFound
 	}
 
-	return nil, ErrQueuedTxIDNotFound
+	return tx, nil
+}
+
+// Set transaction by transaction identifier
+func (q *TxQueue) Set(id common.QueuedTxID, tx *common.QueuedTx) {
+	q.transactions.add(id, tx)
 }
 
 // Remove removes transaction by transaction identifier
 func (q *TxQueue) Remove(id common.QueuedTxID) {
-	q.mu.Lock()
-	defer q.mu.Unlock()
-
-	delete(q.transactions, id)
+	q.transactions.delete(id)
 }
 
 // StartProcessing marks a transaction as in progress. It's thread-safe and
 // prevents from processing the same transaction multiple times.
 func (q *TxQueue) StartProcessing(tx *common.QueuedTx) error {
-	q.mu.Lock()
-	defer q.mu.Unlock()
-
-	if tx.Hash != (gethcommon.Hash{}) || tx.Err != nil {
+	if tx.Hash() != (gethcommon.Hash{}) || tx.Err() != nil {
 		return ErrQueuedTxAlreadyProcessed
 	}
 
-	if tx.InProgress {
+	if tx.InProgress() {
```

Comment:
Before changes, this method properly synchronized processing of a single `QueuedTx` but now it's not. If you run it concurrently, it can happen that one goroutine will execute this check (`tx.InProgress()`) that will return `false` and in another goroutine, the same will happen. Finally, it will be possible to send more than one transaction at the same time. Examine `CompleteTransaction` in `txqueue_manager.go`.

Related to my comment https://github.com/status-im/status-go/pull/456/files#r153428059.